library/ccm.c

```diff
@@ -121,30 +156,55 @@ void mbedtls_ccm_free( mbedtls_ccm_context *ctx )
 /*
  * Update the CBC-MAC state in y using a block in b
  * (Always using b as the source helps the compiler optimise a bit better.)
+ * Initial b masking happens outside of this macro due to various sources of it.
  */
 #define UPDATE_CBC_MAC                                                      \
     for( i = 0; i < 16; i++ )                                               \
-        y[i] ^= b[i];                                                       \
+    {                                                                       \
+        y[perm_table[i]] ^= b[perm_table[i]];                               \
+        y[perm_table[i]] ^= mask_table[perm_table[i]];                      \
+    }                                                                       \
                                                                             \
     if( ( ret = mbedtls_cipher_update( &ctx->cipher_ctx, y, 16, y, &olen ) ) != 0 ) \
         return( ret );
 
+/*
+ * Copy src to dst starting at a random offset, while masking the whole dst buffer.
+ */
+#define COPY_MASK( dst, src, mask, len_src, len_dst )                   \
```

Comment:
Can we save code size if we will define it as function ?

Reply:
I'll check what's the difference.

Reply:
124 bytes saved if we're not introducing any FI protection to the function. 
+33 bytes with added flow control and return value checking.

Reply:
Please note that this increases the code size by 33 bytes.

Reply:
Not sure I understood your point: it is -124 and + 33, so we still save 91 bytes ?

Reply:
or you mean +33 to the original size ?

Reply:
No, it's either:
-124 bytes with the version with no FI and no error codes;
+33 bytes with flow control and return value checking.